### DISCUSSION TITLE
How to get list of files in subdirectory in an archive

### DISCUSSION THREAD

**Discussion by gbook** [rikyoz/bit7z#253] [General]
If I know the relative path within the archive, for example `data/S123/2`, is there a way to get a list of the files within that directory?

**Comment by rikyoz** [comment]
Hi!
Sorry for the late reply.
Unfortunately, there's no direct method for this, at least at the moment.
Obviously, you can achieve the same result by looping over the archive reader and filtering the items, e.g.:
```cpp
constexpr auto prefix = "data/S123/2";
for ( const auto& item : reader ) {
    if ( item.path().rfind( prefix, 0 ) != 0 ) { // Or C++20's starts_with method.
       continue;
    }
    // The current item starts with the prefix
    // ...
}
```

I'll definitely add some convenience methods to achieve the same result with a more clean approach.
For example, I'm thinking of adding a `BitArchiveReader::itemsMatching(...)` method that given a wildcard pattern, it will return a `std::vector` of the items whose relative path within the archive match the pattern.
I hope to add such method in the next v4.0.9 or in the following v4.1-beta.

**Comment by gbook** [comment]
Thank you! I'll try that.
